src/Sylius/Bundle/CoreBundle/Controller/ProductTaxonController.php

```diff
@@ -103,4 +107,27 @@ private function updatePositions(string $position, int $id): void
         $productTaxonFromBase = $this->repository->findOneBy(['id' => $id]);
         $productTaxonFromBase->setPosition((int) $position);
     }
+
+    /**
+     * @return mixed
+     *
+     * @deprecated This function will be removed in Sylius 2.0, since Symfony 5.4, use explicit input sources instead
+     * based on Symfony\Component\HttpFoundation\Request::get
+     */
+    private function getParameterFromRequest(Request $request, string $key)
+    {
+        if ($request !== $result = $request->attributes->get($key, $request)) {
```

Comment:
Why do we have to do it this way? It is not obvious what is the expected result. Why can't we use `has` method?

Reply:
it is a workaround over usage of internal function from `Symfony\Component\HttpFoundation\Request` and it is an almost copy-paste of function `get` from this service. I intentionally don't refactored it, because this is temporary and will be removed in Sylius 2.0